src/humidity.jl

```diff
@@ -0,0 +1,63 @@
+"""
+Compute the saturation specific humidity for a single atmospheric level.
+"""
+function get_saturation_specific_humidity(
+    T::Array{NF,2},  # Absolute temperature for a single atmospheric level
+    p::Array{NF,2},  # Normalised surface pressure
+    σ::NF,           # σ of the current atmospheric level
+    M::ModelSetup{NF},
```

Comment:
Currently `ModelSetup` is an abstract type (so no `NF`). `models.jl` has now `BarotropicModel`, `ShallowWaterModel` and `PrimitiveEquationModel` defined as subtypes of `ModelSetup`, meaning that for precipitation we should write `M::PrimitiveEquationModel` as humidity isn't defined for the other ModelSetups

Reply:
Ah I didn't spot that!

Is there any benefit to making the `ModelSetup` abstract type parametric, i.e. `ModelSetup{NF}`? Probably not, unless you want to dispatch on it somewhere?

Reply:
I don't think there's a need to have `ModelSetup` parametric, making sure we don't mix number formats where we don't intend to happens in lower level functions anyway.